framework/src/main/java/org/checkerframework/framework/stub/RemoveAnnotationsForInference.java

```diff
@@ -0,0 +1,515 @@
+     */
+    public static void main(String[] args) {
+        if (args.length < 1) {
+            System.err.println("Usage: provide one or more directory names to process");
+            System.exit(1);
+        }
+        for (String arg : args) {
+            process(arg);
+        }
+    }
+
+    /** Maps from simple names to fully-qualified names of annotations. */
+    static Multimap<String, String> simpleToFullyQualified = ArrayListMultimap.create();
+
+    static {
+        try {
+            ClassPath cp = ClassPath.from(RemoveAnnotationsForInference.class.getClassLoader());
+            for (ClassPath.ClassInfo ci : cp.getTopLevelClasses()) {
+                simpleToFullyQualified.put(ci.getSimpleName(), ci.getName());
+            }
+        } catch (IOException e) {
+            throw new BugInCF(e);
+        }
+    }
+
+    /**
+     * Process each file in the given directory; see the {@link RemoveAnnotationsForInference class
+     * documentation} for details.
+     *
+     * @param dir directory to process
+     */
+    private static void process(String dir) {
+
+        Path root = JavaStubifier.dirnameToPath(dir);
+
+        RemoveAnnotationsCallback rac = new RemoveAnnotationsCallback();
+        CollectionStrategy strategy = new ParserCollectionStrategy();
+        // Required to include directories that contain a module-info.java, which don't parse by
+        // default.
+        strategy.getParserConfiguration()
+                .setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_11);
+        ProjectRoot projectRoot = strategy.collect(root);
+
+        for (SourceRoot sourceRoot : projectRoot.getSourceRoots()) {
+            try {
+                sourceRoot.parse("", rac);
+            } catch (IOException e) {
+                throw new BugInCF(e);
+            }
+        }
+    }
+
+    /**
+     * Callback to process each Java file; see the {@link RemoveAnnotationsForInference class
+     * documentation} for details.
+     */
+    private static class RemoveAnnotationsCallback implements SourceRoot.Callback {
+        /** The visitor instance. */
+        private final RemoveAnnotationsVisitor rav = new RemoveAnnotationsVisitor();
+
+        @Override
+        public Result process(
+                Path localPath, Path absolutePath, ParseResult<CompilationUnit> result) {
+            Optional<CompilationUnit> opt = result.getResult();
+            if (opt.isPresent()) {
+                CompilationUnit cu = opt.get();
+                List<AnnotationExpr> removals = rav.visit(cu, null);
+                removeAnnotations(absolutePath, removals);
+            }
+            return Result.DONT_SAVE;
+        }
+    }
+
+    /**
+     * Rewrites the file in place, removing the given annotations from it.
+     *
+     * @param absolutePath the path to the file
+     * @param removals the annotations to remove
+     */
+    static void removeAnnotations(Path absolutePath, List<AnnotationExpr> removals) {
```

Comment:
I was confused as to why you weren't using JavaParser to remove the annotations.  This comment:
```
    // An earlier implementation used ModifierVisitor.  However, JavaParser's unparser can change
    // the structure of the program. For example, it changes `protected @Nullable Object x;` to
    // `@Nullable protected Object x;` which yields a type.anno.before.modifier error.
```
This comment explains why.  It would be helpful if it was closer to this code.